lib/iris/fileformats/netcdf.py

```diff
@@ -2104,9 +2194,19 @@ def save(cube, filename, netcdf_format='NETCDF4', local_keys=None,
     with Saver(filename, netcdf_format) as sman:
         # Iterate through the cubelist.
         for cube in cubes:
+            if isinstance(packing, list):
+                if len(packing) > 0:
+                    packspec = packing.pop(0)
+                else:
+                    msg = ('If packing is a list, it must have the '
+                           'same number of elements as the argument to'
+                           'cube.')
+                    raise ValueError(msg)
+            else:
+                packspec = packing
```

Comment:
@jswanljung For this whole section, if we have a `list` (or a `tuple`) we're popping as we go along ... but it can be the case that the user doesn't have the same number of cubes as `packspec`s, and you've dealt with that. Great!

However, we only raise an exception in this case as we stumble across it. From a user perspective, I think I rather see that we **fail early** rather than later.

What I'm suggesting is that before entering the `Saver` context manager, we detect this exception case beforehand, as we already have the `cubes` and the `packing`, so we can easily check that they align in terms of length (associated pairs). If not, then raise the exception. This mean that when we enter the `Saver` we know that we've got a stronger saving contract i.e. we should now save all the `cubes` in the `CubeList`.